src/Host/Client/Impl/Host/RemoteCredentialsDecorator.cs

```diff
@@ -38,10 +37,10 @@ internal class RemoteCredentialsDecorator : ICredentialsDecorator {
             // the first prompt should be validated and saved, and then the same credentials will be reused for the second session.
             var token = await _lock.WriterLockAsync(cancellationToken);
 
-            await _mainThread.SwitchToAsync(cancellationToken);
+            await _coreShell.SwitchToMainThreadAsync(cancellationToken);
```

Comment:
I'd prefer to leave `IMainThread` as a separate service - it is much easier to substitute a single service in tests rather than the whole `ICoreShell`